src/travel-aid/TravelAidScreenComponent.tsx

```diff
@@ -216,29 +230,58 @@ const useTravelAidAnnouncements = (
   state: FocusedEstimatedCallState,
   situationsForFocusedStop: SituationType[],
   noticesForFocusedStop: NoticeFragment[],
+  announcedSituationIds: string[],
+  announcedNoticeIds: string[],
 ) => {
   const {language, t} = useTranslation();
   const isFirstRender = useRef(true);
   const message = getFocussedStateA11yLabel(state, t, language);
-  const situations = getSituationA11yLabel(situationsForFocusedStop, language);
-  const notices = getNoticesA11yLabel(noticesForFocusedStop);
-  const previousQuayId = useRef(state.focusedEstimatedCall.quay.id);
+  const [currentAnnouncedSituationIds, setCurrentAnnouncedSituationIds] =
+    useState<string[]>(announcedSituationIds);
+  const [currentAnnouncedNoticeIds, setCurrentAnnouncedNoticeIds] =
+    useState<string[]>(announcedNoticeIds);
 
   useEffect(() => {
+    const newSituations = situationsForFocusedStop.filter(
+      (s) => !currentAnnouncedSituationIds.includes(s.id),
+    );
+    const newNotices = noticesForFocusedStop.filter(
+      (s) => !currentAnnouncedNoticeIds.includes(s.id),
+    );
+
+    if (newSituations.length > 0) {
+      setCurrentAnnouncedSituationIds((prev) => [
+        ...prev,
+        ...newSituations.map((s) => s.id),
+      ]);
+    }
+
+    if (newNotices.length > 0) {
+      setCurrentAnnouncedNoticeIds((prev) => [
+        ...prev,
+        ...newNotices.map((s) => s.id),
+      ]);
+    }
+
     if (isFirstRender.current) {
       isFirstRender.current = false;
       return;
     }
 
-    if (state.focusedEstimatedCall.quay.id !== previousQuayId.current) {
-      previousQuayId.current = state.focusedEstimatedCall.quay.id;
-      AccessibilityInfo.announceForAccessibility(
-        message + screenReaderPause + notices + screenReaderPause + situations,
-      );
-    } else {
-      AccessibilityInfo.announceForAccessibility(message);
-    }
-  }, [message, situations, notices, state.focusedEstimatedCall.quay.id]);
+    const situations = getSituationA11yLabel(newSituations, language);
+    const notices = getNoticesA11yLabel(newNotices);
+    AccessibilityInfo.announceForAccessibility(
+      message + screenReaderPause + situations + screenReaderPause + notices,
+    );
+  }, [
+    message,
+    state.focusedEstimatedCall.quay.id,
+    currentAnnouncedSituationIds,
+    currentAnnouncedNoticeIds,
+    noticesForFocusedStop,
+    situationsForFocusedStop,
+    language,
+  ]);
```

Comment:
Such large dependency lists are quite risky, and it also seems that some of the elements here are not referentially stable between renders? Just by reading the code I would be a little scared that something is announced on every render cycle. 

If possible, try to create the message outside of the `useEffect`, and let the effect be stripped down to just what it should actually do: Announce a message.

Reply:
Sure, I going to refactor a bit more this part.